src/Common/src/System/Net/Mail/MailAddressParser.cs

```diff
@@ -309,7 +309,18 @@ private static string ParseDisplayName(string data, ref int index, bool expectMu
 
         internal static string NormalizeOrThrow(string input)
         {
+#if NETSTANDARD1_7
+            try
+            {
+                return input.Normalize(Text.NormalizationForm.FormC);
+            }
+            catch (ArgumentException e)
+            {
+                throw new FormatException(SR.MailAddressInvalidFormat, e);
+            }
+#else
```

Comment:
Why is the else necessary? i.e. why can't this just be the code you've added, without the if/else?